bin/report

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# bin/report <build-dir> <cache-dir> <env-dir>
+
+### Configure environment
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+BUILD_DIR=${1:-}
+
+# <separator> <array>
+joinArrayWithSeparator() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+packages=()
+custom_packages=()
+custom_repositories=()
+
+while IFS= read -r line; do
+  if [[ $line ]]; then
+    if echo "${line}" | grep -q -e "^:repo:"; then
+      custom_repositories+=("${line//:repo:/}")
+    elif [[ $line == *deb ]]; then
+      custom_packages+=("${line}")
+    else
+      packages+=("${line}")
+    fi
+  fi
+done < <(grep -v -s -e '^#' < "${BUILD_DIR}/Aptfile")
+
+echo "packages: \"$(joinArrayWithSeparator "," "${packages[@]}")\""
```

Comment:
I wouldn't wrap this output in double quotes if you're combining the packages using just a comma - otherwise, technically, it's one single string, right?

Reply:
i copied the output format from the node buildpack which outputs lines like the following:

```
package_manager: "npm"
start_script: "node foo.js"
```

so i was following a similar pattern to output the following:

```
packages: "libexample-dev,libexample-dev2"
custom_packages: "https://downloads.example.com/example.deb"
custom_repositories: "deb https://apt.example.com/ example-distro main"
```

Reply:
Hmh, I see... should the strings at least be escaped then, just in case somebody manages to sneak in a double quote somewhere?

It might make sense to sort the list of packages, too, wdyt?

Reply:
yeah, sorting is a good addition.  i'll also escape the strings just to be safer.

Reply:
done. i put all the logic for sorting, joining, and escaping into the `output_key_value` function which also follows what the Node.js buildpack does and skips the output if the value is empty.